server/plugin/utils.go

```diff
@@ -343,3 +343,19 @@ func (p *Plugin) GetSubscriptionsForAccessibleChannelsOrProjects(subscriptionLis
 
 	return filteredSubscriptionList, nil
 }
+
+func (p *Plugin) VerifyEncryptedWebhookSecret(received string) (status int, err error) {
+	decodedWebhookSecret, err := p.Decode(received)
+	if err != nil {
+		return http.StatusInternalServerError, errors.New("failed to decode webhook secret")
+	}
+	decryptedWebhookSecret, err := p.Decrypt(decodedWebhookSecret, []byte(p.getConfiguration().EncryptionSecret))
+	if err != nil {
+		return http.StatusInternalServerError, errors.New("failed to decrypt webhook secret")
+	}
+	if p.getConfiguration().WebhookSecret != string(decryptedWebhookSecret) {
```

Comment:
Use something like this - https://github.com/Brightscout/mattermost-plugin-servicenow/blob/953125b99019304f398e30bc2f7e8df3aeba1438/server/plugin/api.go#L149

Reply:
cannot use that directly here as we are receiving as query param and not URL encoded FORM

Reply:
I meant this - https://github.com/Brightscout/mattermost-plugin-servicenow/blob/953125b99019304f398e30bc2f7e8df3aeba1438/server/plugin/api.go#L824

Reply:
Tried it, escaped URL query param is not supported by the Azure DevOps, I checked further it gives error as
`failed to create subscription: errorMessage Subscription input 'url' does not match the required data type 'Uri'.`

Reply:
As discussed we don't need this, so resolving it